lib/cloud/clients.go

```diff
@@ -453,6 +457,17 @@ func (c *cloudClients) GetGCPGKEClient(ctx context.Context) (gcp.GKEClient, erro
 	return c.initGCPGKEClient(ctx)
 }
 
+// GetGCPInstancesClient returns instances client.
+func (c *cloudClients) GetGCPInstancesClient(ctx context.Context) (gcp.InstancesClient, error) {
+	c.mtx.RLock()
+	if c.gcpInstances != nil {
+		defer c.mtx.RUnlock()
+		return c.gcpInstances, nil
+	}
+	c.mtx.RUnlock()
+	return c.initGCPInstancesClient(ctx)
```

Comment:
Instead of doing all that `Lock/Unlock` thing can you use `sync.Once` to make sure that this object is initialized only once? Then `initGCPInstancesClient` can be simplified as locking won't be required there.